src/ir_BluestarHeavy.cpp

```diff
@@ -0,0 +1,150 @@
+// Copyright 2020 David Conran (crankyoldgit)
+/// @file
+/// @brief Support for BluestarHeavy protocol
+
+// Supports:
+//   Brand: Bluestar,  Model: TODO add device and remote
+
+#include "IRrecv.h"
+#include "IRsend.h"
+#include "IRutils.h"
+
+// WARNING: This probably isn't directly usable. It's a guide only.
+
+// See https://github.com/crankyoldgit/IRremoteESP8266/wiki/Adding-support-for-a-new-IR-protocol
+// for details of how to include this in the library.
+const uint16_t kBluestarHeavyHdrMark = 4912;
+const uint16_t kBluestarHeavyBitMark = 465;
+const uint16_t kBluestarHeavyHdrSpace = 5058;
+const uint16_t kBluestarHeavyOneSpace = 1548;
+const uint16_t kBluestarHeavyZeroSpace = 572;
+const uint16_t kBluestarHeavyFreq = 38000;  // Hz. (Guessing the most common frequency.)
+//const uint16_t kBluestarHeavyBits = 104;  // Move to IRremoteESP8266.h
+//const uint16_t kBluestarHeavyStateLength = 13;  // Move to IRremoteESP8266.h
+const uint16_t kBluestarHeavyOverhead = 3;
+// DANGER: More than 64 bits detected. A uint64_t for 'data' won't work!
+
+
+#if SEND_BLUESTARHEAVY
+// Alternative >64bit function to send BLUESTARHEAVY messages
+// Function should be safe over 64 bits.
+/// Send a BluestarHeavy formatted message.
+/// Status: ALPHA / Untested.
+/// @param[in] data An array of bytes containing the IR command.
+///                 It is assumed to be in MSB order for this code.
+/// e.g.
+/// @code
+///   uint8_t data[kBluestarHeavyStateLength] = {0xD5, 0xFE, 0xD7, 0x4F, 0xFA, 0x5F, 0xFA, 0x5F, 0xFF, 0x7F, 0x5C, 0xFD, 0xDC};
+/// @endcode
+/// @param[in] nbytes Nr. of bytes of data in the array. (>=kBluestarHeavyStateLength)
+/// @param[in] repeat Nr. of times the message is to be repeated.
+void IRsend::sendBluestarHeavy(const uint8_t data[], const uint16_t nbytes, const uint16_t repeat) {
+  for (uint16_t r = 0; r <= repeat; r++) {
+    uint16_t pos = 0;
+    // Data Section #2
+    // e.g.
+    //   bits = 104; bytes = 13;
+    //   *(data + pos) = {0xD5, 0xFE, 0xD7, 0x4F, 0xFA, 0x5F, 0xFA, 0x5F, 0xFF, 0x7F, 0x5C, 0xFD, 0xDC};
+    sendGeneric(kBluestarHeavyHdrMark, kBluestarHeavyHdrSpace,
+                kBluestarHeavyBitMark, kBluestarHeavyOneSpace,
+                kBluestarHeavyBitMark, kBluestarHeavyZeroSpace,
+                kBluestarHeavyHdrMark, kDefaultMessageGap,
+                data + pos, nbytes,  // Bytes
+                kBluestarHeavyFreq, true, kNoRepeat, kDutyDefault);
+    pos += nbytes;  // Adjust by how many bytes of data we sent
+  }
```

Comment:
This can be simplified to:
```c++
  sendGeneric(kBluestarHeavyHdrMark, kBluestarHeavyHdrSpace,
                kBluestarHeavyBitMark, kBluestarHeavyOneSpace,
                kBluestarHeavyBitMark, kBluestarHeavyZeroSpace,
                kBluestarHeavyHdrMark, kDefaultMessageGap,
                data, nbytes,  // Bytes
                kBluestarHeavyFreq, true, repeat, kDutyDefault);
```
I.e. No `for` loop needed.